controller/api/destination/server.go

```diff
@@ -364,17 +364,44 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	return nil
 }
 
-func (s *server) createAddress(pod *corev1.Pod, port uint32) (watcher.Address, error) {
+// getPortForPod returns the port that a `pod` is listening on.
+//
+// If the `host` address given is the HostIP, use the HostPort mapping for the `pod`
+// to determine the port to return.
+func (s *server) getPortForPod(pod *corev1.Pod, host string, port uint32) (uint32, error) {
+	if host == pod.Status.PodIP {
+		return port, nil
+	}
+
+	if host == pod.Status.HostIP {
+		for _, container := range pod.Spec.Containers {
+			for _, containerPort := range container.Ports {
+				if uint32(containerPort.HostPort) == port {
+					return uint32(containerPort.ContainerPort), nil
+				}
+			}
+		}
+	}
+
+	return 0, fmt.Errorf("unable to find container port as host (%s) matches neither PodIP nor HostIP (%s)", host, pod)
+}
+
+func (s *server) createAddress(pod *corev1.Pod, host string, port uint32) (watcher.Address, error) {
 	ownerKind, ownerName := s.k8sAPI.GetOwnerKindAndName(context.Background(), pod, true)
+	port, err := s.getPortForPod(pod, host, port)
```

Comment:
Is this correct?

```suggestion

    // Proxies usually receive traffic targeting `podIp:containerPort`.
    // However, they may looking up traffic  `nodeIp:nodePort`. In this
    // case, we convert the port to the containerPort for discovery.
	port, err := s.getPortForPod(pod, host, port)
```

Reply:
Yes and I moved that comment into the doc for `getPortForPod` (who only gets called here so it's kind of a toss-up as to where the comment should go)